src/rocksdb.rs

```diff
@@ -1856,6 +1856,10 @@ impl DB {
         DB::open_default(restore_db_path)
     }
 
+    pub fn checkpoint(&self) -> Result<crate::Checkpoint, String> {
```

Comment:
I think RocksDB's name is confusing. `Checkpointer` and `new_checkpointer` is better.